src/pybind/mgr/cephadm/ssh.py

```diff
@@ -188,13 +188,16 @@ async def _write_remote_file(self,
                                  ) -> None:
         try:
             dirname = os.path.dirname(path)
-            await self._check_execute_command(host, ['mkdir', '-p', dirname], addr=addr)
-            tmp_path = path + '.new'
+            await self._check_execute_command(host, ['mkdir', '-p', '/tmp' + dirname], addr=addr)
+            tmp_path = '/tmp' + path + '.new'
             await self._check_execute_command(host, ['touch', tmp_path], addr=addr)
             if uid is not None and gid is not None and mode is not None:
                 # shlex quote takes str or byte object, not int
                 await self._check_execute_command(host, ['chown', '-R', str(uid) + ':' + str(gid), tmp_path], addr=addr)
                 await self._check_execute_command(host, ['chmod', oct(mode)[2:], tmp_path], addr=addr)
+            elif self.mgr.ssh_user != 'root':
+                await self._check_execute_command(host, ['chown', '-R', self.mgr.ssh_user, tmp_path], addr=addr)  # type: ignore
```

Comment:
why do you need `type: ignore`?

Reply:
without `type: ignore`, from mypy on line 199, it says that `self.mgr.ssh_user` is type optional str, but `_check_execute_command` is expecting type str. But I thought `self.mgr.ssh_user` should already be set to something by that time since we do `assert self.mgr.ssh_user` in `_remote_connection`

Reply:
right the assert is mainly for the type checker to be happy. (you can avoid that by making sure embers are always initialized and never default to None. I'd encourage this style, but it's not always possible).

The problem with type: ignore is that this ignores all type errors, whereas `assert self.mgr.ssh_user` only reduces the type of this variable. I'd encourage to make the type checker happy instead of adding ignores.

Reply:
right, makes sense. I'll try to avoid adding ignores in the future and this time use assert instead